docs/DeveloperGuide.md

```diff
@@ -7,32 +7,97 @@
+|v2.1|Expert user|Add in countries to the list of available countries|Update the itinerary|
+|v2.1|Expert user|Sort all reservations according to month|Analyze trends throughout the year|
+|v2.1|Expert user|Mark a reservation as ‘completed’|Access both completed and ongoing reservations|
+|v2.1|Expert user|Compute the price of custom travel packages|Get a rough estimate for what to charge the customer|
 
 ## Non-Functional Requirements
 
-{Give non-functional requirements}
+### Performance Requirements
+- Should work on any mainstream OS as long as it has Java 11 or above installed.
+
+### Usability
+- A user with above average typing speed for regular English text (i.e. not code, not system admin commands) should be able to accomplish most of the tasks faster using commands than using the mouse.
+- Prompts and instructions must be in proper and concise sentences for ease of understanding.
 
 ## Glossary
 
 * *glossary item* - Definition
+* Packages - Travel Package within the agency's database
+* Reservations - Reservation of travel package made by one customer through the app 
 
 ## Instructions for manual testing
 
 {Give instructions on how to do a manual product testing e.g., how to load sample data to be used for testing}
+
+### Summary of User Commands
+
+| Command | Format Examples |
+| ------- | --------------- |
+|packages|packages|
+|info|info {num} (num < number of available packages) <br/> e.g. info 2 |
+|add|add {package_name} {country} {duration} {price} {vacancies} <br/>  e.g. add Skiing_Trip Sweden 15/2/2022-19/2/2022 800 100|
+|delete|delete {num} (num < number of available packages) <br/> e.g. delete 2|
+|reserve|reserve {package_number} {contact_name} {contact_number} {number_pax} <br/> e..g reserve 3 John 91234567 3 |
+|remove|remove {reservation_id} <br/> e..g remove R001|
+|reservations|reservations {package_number} <br/> eg. reservations 2|
+
+## Feature - Help Command 
+Aim: Displays a list of all available commands that the user can refer to as a guide
+
+## Feature - Storage
+#### Initialisation (Loading Data)
+The sequence diagram that shows how `Storage` is created and the data is loaded from the saved files when the program is initialised is shown below: </br>
+![](StorageSeqDiag.png)
+
+1. `TARBS` creates a Storage object with the relevant file names.
+2. `TARBS` then calls the `createPackages()` method of the Storage class
+3. `storage` will then load the contents of the file in the `Reservation` and `TravelPackages` file calling the different `parse` methods. The exact implementation is not shown in the diagram.
+4. A new `Package` object is constructed with the relevant data.
+5. `storage` returns `package` object to `TARBS`. 
+
+## Feature - Parser
+The sequence diagram shows an example of how a user input is parsed and returns a new `Command`.
+
+For simplicity, `ByeCommand` is shown here:
+
+![image](https://user-images.githubusercontent.com/53790951/160289728-5a17b2ea-0599-48f4-a18b-cadfbbcbfb11.png)
```

Comment:
![image](https://user-images.githubusercontent.com/74888935/160959015-bb5dba17-de1b-4f3a-97ed-4ccb31c248e1.png)
Similar to the storage diagram, maybe the highlighted line should be a solid line